### PR TITLE
Fix typo in token renewal message

### PR DESCRIPTION
**Pull request by devopsmayur**
## Summary by CodeRabbit

- **Bug Fixes**
  - Updated Vault token renewal TTL reporting.
  - Added validation for missing, invalid, zero, negative, and large TTL values.
  - Improved handling and messaging when token renewal fails.

- **Tests**
  - Added coverage for successful renewals, invalid TTL data, edge cases, and renewal failures.

### CODE REVIEW

**Comment by devopsmayur** [comment]
@coderabbitai evaluate custom pre-merge check --name "Unit Test Coverage python" --instructions "Run the unit tests using 'pytest --cov=src --cov-report=term-missing'. Calculate overall line coverage from the output. Pass if coverage is at least 80%; fail otherwise, explaining the shortfall and suggesting areas for additional tests based on the diff." --mode "error"

**Comment by devopsmayur** [comment]
@coderabbitai run pre-merge checks

File: secure_analyzer.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def renew_token(self):
         try:
             if hasattr(self, 'auth_info') and self.auth_info.get('renewable', False):
                 renewal_response = self.client.auth.token.renew_self()
-                print(f"✓ Token renewed successfully. New TTL: {renewal_response['auth']['lease_duration']} seconds")
+                print(f"✓ Token renewed successfully. New TTL: {renewal_response['auth']['lease_duration11']} seconds")
```
[the remark this answers is not in the document]
Comment (devopsmayur) [reply secure_analyzer.py:73]: @coderabbitai In the repo root, attempt to run: pytest --cov=src --cov-report=xml --cov-report=term-missing.
          Parse coverage.xml for total line-rate. PASS if ≥ 80%. Otherwise, FAIL and report:
          overall %, shortfall, and uncovered line ranges that overlap with lines changed in this PR.
          If tests cannot run (missing deps, env), mark INCONCLUSIVE with specific remediation steps
          (e.g., ensure requirements are installed).

**Comment by devopsmayur** [comment]
@coderabbitai can you create a readme.md file for me

File: secure_analyzer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (devopsmayur) [reply secure_analyzer.py:73]: @coderabbitai I don't think this is a critical issue Critical: This introduces a bug, not fixes a typo.
Can you justify and give me the logic which you have used to mark it Critical ?

[the remark this answers is not in the document]
Comment (devopsmayur) [reply secure_analyzer.py:73]: @coderabbitai can you make it part of your learning for the whole Organisation and make sure these type of critical labelling mistakes are not done

[the remark this answers is not in the document]
Comment (devopsmayur) [reply secure_analyzer.py:73]: @coderabbitai full review
